docs/ops/doc/ParameterConversion.rst

```diff
@@ -0,0 +1,141 @@
+    var result = ops.op("filter.convolve").input(in, kernel).apply();
+
+The only step for us as the developer is to tell SciJava Ops that it can convert ``double[][]``\ s to ``RandomAccessibleInterval<DoubleType>``\ s, which we do with ``engine.convert`` Ops.
+
+A simple ``engine.convert`` Op
+==============================
+
+All ``engine.convert`` Ops are simple ``Function``\ s, that take as input the user argument to the Op, and return a *translation* of that data into the type expected by Ops. In our case, we want to convert *from* the user's ``double[][]`` into a ``RandomAccessibleInterval<DoubleType>``:
+
+.. code-block:: java
+
+    /**
+     * @param input the input data
+     * @return an output image whose values are equivalent to {@code input}s
+     *         values but whose element types are {@link BitType}s.
+     * @implNote op names='engine.convert', type=Function
+     */
+    public static RandomAccessibleInterval<DoubleType> arrayToDoubles(final double[][] input)
+    {
+        // Creates an empty image of doubles
+        var img = ArrayImgs.doubles(input.length, input[0].length);
+        var ra = img.randomAccess();
+        // Deep copies the double[][] into the RAI
+        for(int i = 0; i < input.length; i++) {
+            for(int j = 0; j < input[0].length; j++) {
+                ra.setPositionAndGet(i, j).set(input[i][j]);
+            }
+        }
+        return img;
+    }
+
+This Op, discovered through the SciJava Ops Indexer, is **all** that is needed to make the execution pattern we want functional.
+
+
+Adding efficiency
+=================
+
+While the above ``engine.convert`` Op is *functional*, it may not be *fast* as the data size increases. This is due to the **copy** inherent in its execution, as the ``ArrayImg`` contains new data structures.
+
+In such cases, devising methods to *wrap* the user arguments, instead of *copying* it, will maximize performance and wow your users. In our case, we can refine our ``engine.convert`` Op to wrap user data, using the ``DoubleAccess`` interface of ImgLib2:
+
+.. code-block:: java
+
+	/**
+	 * @param input the input data
+	 * @return an output image whose values are equivalent to {@code input}s
+	 *         values but whose element types are {@link BitType}s.
+	 * @implNote op names='engine.convert', type=Function
+	 */
+	public static RandomAccessibleInterval<DoubleType> arrayToDoubles(final double[][] input)
+	{
+		// Wrap 2D array into DoubleAccess usable by ArrayImg
+		var access = new DoubleAccess() {
+
+			private final int rowSize = input[0].length;
+
+			@Override
+			public double getValue(int index) {
+				var row = index / rowSize;
+				var col = index % rowSize;
+				return input[row][col];
+			}
+
+			@Override
+			public void setValue(int index, double value) {
+				var row = index / rowSize;
+				var col = index % rowSize;
+				input[row][col] = value;
+			}
+		};
+		return ArrayImgs.doubles(access, input.length, input[0].length);
```

Comment:
This is very slick :sunglasses:

Reply:
Yeah, I was really happy with it - only issue is that in practice it would fail for `double[][]`s with many elements due to too many positions.